### DISCUSSION TITLE
candidates and chromosomes

### DISCUSSION THREAD

**Discussion by huahuav587** [KRM7/gapp#8] [General]
How can I look at the candidates for each generation of the population and their chromosomes？

**Comment by KRM7** [comment]
The old populations are normally discarded after each generation, but you can collect them using a callback function,
for example:

```cpp
std::vector<Population<RealGene>> populations;

RCGA rcga;
rcga.on_generation_end([&](const GaInfo& ga)
{
    const auto& population = static_cast<const RCGA&>(ga).population();
    populations.push_back(population);
});

// rcga.solve(...);
```

**Reply by huahuav587** [reply]
OK,thank you so much!
